golang/cosmos/x/swingset/genesis.go

```diff
@@ -131,7 +182,17 @@ func (eventHandler swingStoreGenesisEventHandler) OnExportRetrieved(provider kee
 
 	artifactsProvider := keeper.SwingStoreExportProvider{
 		GetExportDataReader: func() (agoric.KVEntryReader, error) {
-			return nil, nil
+			exportDataIterator := eventHandler.swingStore.Iterator(nil, nil)
+			kvReader := agoric.NewKVIteratorReader(exportDataIterator)
+			eventHandler.hasher.Reset()
```

Comment:
Why does the hasher need to be reset here? Is it used more than once? I want to make sure we don't have a bug where we only actually hash the last bit of the data, because we reset it on every loop or something.

Reply:
Technically `GetExportDataReader` could be called more than once

Reply:
oh, I had assumed that the hasher was being created one-per-export, but maybe the `eventHandler` that it lives in is allocated statically, one-per-process?

Reply:
the artifact provided is passed to `WriteSwingStoreExportToDirectory`. That function is well behaved and will only read the stream export data once, but technically it could do it multiple times.

I agree the layering is weird, and in a perfects world I'd decouple the generation of the hash from the consumer that writes the data to disk, however I wanted to avoid that inefficiency. I suppose I could break boundaries, grab the iterator once on the outer scope, assert we only wrap the iterator once, and at the end verify that the iterator has been fully consumed.